presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java

```diff
@@ -192,8 +205,22 @@ public void queryCompletedEvent(QueryInfo queryInfo)
             }
 
             Optional<String> plan = Optional.empty();
-            if (queryInfo.getPlan().isPresent()) {
-                plan = Optional.of(objectMapper.writeValueAsString(queryInfo.getPlan().get()));
+            try {
+                if (queryInfo.getOutputStage().isPresent()) {
+                    Session session = queryInfo.getSession().toSession(sessionPropertyManager);
+                    plan = Optional.of(textDistributedPlan(
+                            queryInfo.getOutputStage().get(),
+                            functionRegistry,
+                            new ComposableStatsCalculator(ImmutableList.of()),
+                            new CostCalculatorUsingExchanges(() -> 1),
```

Comment:
if you want to suppress costing, you could pass `(....) -> UNKNOWN_COST` here. Same for stats calculator.

Also, please add a comment why this is so, e.g. `// Suppress stats & costs, since transaction is already completed`